### PR TITLE
feat: add grafana on call as chat provider

### LINKED ISSUE #4682: 🚀 Feature: Add Grafana On-call as a Incident Management Provider
### 🔖 Feature description

Use the [Grafana On-call docs](https://grafana.com/docs/oncall/latest/integrations/webhook/) to create a new push provider while following [Novu's guide to a new provider](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider)

This provider should act functionally similar to chat provider in that the credentials will be passed by a subscriber (alert_uid, state, etc) except for the message and title.

### 🎤 Why is this feature needed ?

more options for the community

### ✌️ How do you aim to achieve this?

community contribution

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by mahendraHegde**
### What change does this PR introduce?

- [x] Adds Grafana On call as a new chat provider
- [x] Adds relavant test cases
- [ ] Updates provider docs 

### Why was this change needed?
closes #4682  and  #4438

#### Integration config
Screenshot 2023-11-04 at 10 52 47 AM

#### Workflow
Screenshot 2023-11-04 at 10 53 29 AM

Screenshot 2023-11-04 at 10 54 15 AM

#### Grafana dashboard
Screenshot 2023-11-04 at 10 55 59 AM
Screenshot 2023-11-04 at 10 56 12 AM

### CODE REVIEW

File: providers/grafana-on-call/package.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+  "name": "@novu/grafana-on-call",
+  "version": "0.16.3",
```
Comment (Cliftonz) [inline providers/grafana-on-call/package.json:3]: This version should be the same as the rest of the providers, v0.21.0

File: providers/grafana-on-call/src/lib/grafana-on-call.provider.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      image_url: this.config.imageUrl,
+      state: this.config.state,
+      link_to_upstream_details: this.config.externalLink,
+      message: options.content,
+    };
+    //response is just string "Ok."
+    const { headers } = await this.axiosInstance.post(url.toString(), body);
+
+    return {
+      id: this.config.alertUid || options.content,
```
Comment (Cliftonz) [inline providers/grafana-on-call/src/lib/grafana-on-call.provider.ts:39]: What does the options.content return?

**Review by Cliftonz — CHANGES_REQUESTED** [review]
just a few questions and changes

File: providers/grafana-on-call/src/lib/grafana-on-call.provider.ts (hunk 2 above)
Comment (mahendraHegde) [reply providers/grafana-on-call/src/lib/grafana-on-call.provider.ts:39]: this is actually the message itself, there is no valid identifier that grafana returns in response(body/header), it it acceptable to generate a uuid and use it as it instead?

File: providers/grafana-on-call/package.json (hunk 1 above)
Comment (mahendraHegde) [reply providers/grafana-on-call/package.json:3]: okay, out of curiocity, whats the reason to keep all provider version same? also this is auto generated.

Comment (Cliftonz) [reply providers/grafana-on-call/package.json:3]: I am not sure, that is a @LetItRock question

File: providers/grafana-on-call/src/lib/grafana-on-call.provider.ts (hunk 2 above)
Comment (Cliftonz) [reply providers/grafana-on-call/src/lib/grafana-on-call.provider.ts:39]: For this use-case yes, we should not be using the actual message as the id here.

File: providers/grafana-on-call/package.json (hunk 1 above)
Comment (mahendraHegde) [reply providers/grafana-on-call/package.json]: upgraded to 0.21.0

File: providers/grafana-on-call/src/lib/grafana-on-call.provider.ts (hunk 2 above)
Comment (mahendraHegde) [reply providers/grafana-on-call/src/lib/grafana-on-call.provider.ts]: done

File: apps/web/public/static/images/providers/dark/square/grafana-on-call.svg
Comment (michaldziuba03) [inline apps/web/public/static/images/providers/dark/square/grafana-on-call.svg:1]: I would suggest to resize square logo svgs to 32x32

**Review by Cliftonz — APPROVED** [review]
Looks good, just need to fix the icon.

File: apps/web/public/static/images/providers/dark/square/grafana-on-call.svg
Comment (mahendraHegde) [reply apps/web/public/static/images/providers/dark/square/grafana-on-call.svg:1]: done.

**Pull request merged**